src/generic_factory.h

```diff
@@ -985,4 +986,65 @@ typename std::enable_if<std::is_constructible<T, std::string>::value, bool>::typ
     return res;
 }
 
+inline bool assign( JsonObject &jo, const std::string &name, units::volume &val,
+                    bool strict = false,
+                    const units::volume lo = units::volume( std::numeric_limits<units::volume::value_type>::min(),
+                            units::volume::unit_type{} ),
+                    const units::volume hi = units::volume( std::numeric_limits<units::volume::value_type>::max(),
+                            units::volume::unit_type{} ) )
+{
+    // Currently JSON data contains volume in 250ml units.
+    // TODO: change JSON to contain milliliter values.
```

Comment:
> Do you mean to use /// instead of plain // so Doxygen extracts this?

Or even better `@todo`

> Yes, but external mods will silently break (items will have a wrong volume).

We can update mods that are distributed as part of the project in the same manner. External mods are frequently subject to arbitrary breakage in any case. Having two competing systems would be a significant problem - this PR as it stands is fine but the JSON switchover should be 100%